FABulous/fabric_generator/fabric_gen.py

```diff
@@ -1990,69 +2000,118 @@ def generateFabric(self) -> None:
-                        # prefix for super tile port
-                        if superTile:
-                            pre = f"Tile_X{i}Y{j}_"
-                        else:
-                            pre = ""
-                        # frameStrobe signal
-                        if y + 1 >= self.fabric.numberOfRows:
+                            done = True
+                            break
+
+                        # No non-NULL tile was found, and tile is not part of a supertile.
+                        # Connect to the fabrics Tile_Y{y}_FrameData signals.
+                        if not done:
                             portsPairs.append(
-                                (f"{pre}FrameStrobe", f"Tile_X{x}_FrameStrobe")
+                                (f"{pre}FrameData", f"Tile_Y{supertile_y}_FrameData")
                             )
 
-                        elif (
-                            y + 1 < self.fabric.numberOfRows
-                            and self.fabric.tile[y + 1][x] == None
+                        # Connecting FrameData_O is easier:
+                        # Always connect FrameData_O, except the next tile in the row is
+                        # part of the supertile (already connected internally).
+                        if (supertile_x + 1, supertile_y) not in superTileLoc:
+                            portsPairs.append(
+                                (
+                                    f"{pre}FrameData_O",
+                                    f"Tile_X{supertile_x}Y{supertile_y}_FrameData_O",
+                                )
+                            )
+
+                        # Connect the FrameStrobe port to the previous tiles'
+                        # FrameStrobe_O signals.
+                        # If the previous tile is NULL, continue the search.
+                        # If all previous tiles are NULL, connect to the fabrics
+                        # Tile_X{x}_FrameStrobe signals.
+
+                        done = False
+
+                        # Get all y-positions before this tile
+                        # Note: the FrameStrobe signals come from the bottom of the
+                        #       fabric, therefore count upwards
+                        for search_y in range(
+                            supertile_y + 1, self.fabric.numberOfRows
                         ):
+
+                            # Previous tile is part of the same supertile.
+                            # FrameStrobe signals are connected internally.
+                            # Stop the search and be done.
+                            if (supertile_x, search_y) in superTileLoc:
+                                done = True
+                                break
+
+                            # Previous tile is NULL, continue search
+                            if self.fabric.tile[search_y][supertile_x] == None:
+                                print("skip")
```

Comment:
Can put to `logger.debug`

Reply:
Whoops, that was left over from testing. I'll remove it.